docs/build.md

```diff
@@ -0,0 +1,120 @@
+    imagePullSecrets: 
+    - name: builder-secret  
+  cacheName: persisent-volume-claim-name
+  source:
+    git:
+      url: https://github.com/buildpack/sample-java-app.git
+      revision: master
+  env:
+  - name: "JAVA_BP_ENV"
+    value: "value"
+  resources:
+    limits:
+      cpu: "0.25"
+      memory: "128M"
+    requests:
+      cpu: "0.5"
+      memory: "256M"
+```
+
+- `tags`: A list of docker tags to build. At least one tag is required.
+- `serviceAccount`: The Service Account name that will be used for credential lookup. Check out the [secrets documentation](secrets.md) for more information. 
+- `builder.image`: This is the tag to the [Cloud Native Buildpacks builder image](https://buildpacks.io/docs/using-pack/working-with-builders/) to use in the build. Unlike on the Image resource, this is an image not a reference to a Builder resource.    
+- `builder.imagePullSecrets`: An optional list of pull secrets if the builder is in a private registry. [To create this secret please reference this link](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials)
+- `source`: The source location that wil be the input to the build. See the [Source Configuration](#source-config) section below.
+- `cacheName`: Optional name of a persistent volume claim to used for a build cache across builds.
+- `env`: Optional list of build time environment variables.
+- `resources`: Optional configurable resource limits on `CPU` and `memory`.
+
+> Note: All fields on a build are immutable. Instead of updating a build, create a new one.
+ 
+##### <a id='source-config'></a>Source Configuration
+
+The `source` field is a composition of a source code location and a `subpath`. It can be configured in exactly one of the following ways:
+
+* Git
+
+    ```yaml
+    source:
+      git:
+        url: ""
+        revision: ""
+      subPath: ""
+    ```
+    - `git`: (Source Code is a git repository)
+        - `url`: The git repository url. For now, only https repositories are supported.
+        - `revision`: The git revision to use. This value may be a commit sha, branch name, or tag.
+    - `subPath`: A subdirectory within the source folder where application code resides. Can be ignored if the source code resides at the `root` level.
+
+* Blob
+
+    ```yaml
+    source:
+      blob:
+        url: ""
+      subPath: ""
+    ```
+    - `blob`: (Source Code is a blob/jar in a blobstore)
+        - `url`: The URL of the source code blob. This blob needs to either be publicly accessible or have the access token in the URL
+    - `subPath`: A subdirectory within the source folder where application code resides. Can be ignored if the source code resides at the `root` level.
+
+* Registry
+
+    ```yaml
+    source:
+      registry:
+        image: ""
+        imagePullSecrets:
+        - name: ""
+      subPath: ""
+    ```
+    - `registry` ( Source code is an OCI image in a registry)
```

Comment:
Does htis have to be an OCI image?  Can we build from a docker v1 image as well?

Reply:
I don't believe that we can.